### PR TITLE
BUGFIX: #2962 JsonArrayType Regression

### LINKED ISSUE #2962: BUG: Regression private const FLOW_JSON_ARRAY
Regression from #2900

The use of private results in a regression:

```
➜  neos-manufacture-highest git:(8.3) ✗ ./flow site:import Neos.Demo   
This command can use a lot of memory when importing sites with many resources.
If the import is successful, you will see a message saying "Import of site ... finished".
If you do not see this message, the import failed, most likely due to insufficient memory.
Increase the memory_limit configuration parameter of your php CLI to attempt to fix this.
Starting import...
---
Cannot access private constant Neos\Flow\Persistence\Doctrine\DataTypes\JsonArrayType::FLOW_JSON_ARRAY

  Type: Error
  File: Data/Temporary/Development/Cache/Code/Flow_Object_Classes/Neos_ContentRepos
        itory_Domain_Service_ImportExport_NodeImportService.php
  Line: 642
```

_Originally posted by @mhsdesign in https://github.com/neos/flow-development-collection/pull/2900#discussion_r1114164545_

Workaround: Remove the `private` :D

### PR DESCRIPTION
**Pull request by mhsdesign**
Fixes: #2962

### CODE REVIEW

**Review by markusguenther — APPROVED** [review]

**Comment by mhsdesign** [comment]
Will repair the Neos8.3 tests… it seems there are no tests in flow that should fail …

File: Neos.Flow/Classes/Persistence/Doctrine/DataTypes/JsonArrayType.php [hunk 1]
```diff
@@ -27,7 +27,7 @@
  */
 class JsonArrayType extends JsonType
 {
-    private const FLOW_JSON_ARRAY = 'flow_json_array';
+    const FLOW_JSON_ARRAY = 'flow_json_array';
```
Comment (mhsdesign) [inline Neos.Flow/Classes/Persistence/Doctrine/DataTypes/JsonArrayType.php]:
```suggestion
    protected const FLOW_JSON_ARRAY = 'flow_json_array';
```

**Comment by mficzel** [comment]
@mhsdesign since this was merged today the Neos 8.3 branches are now red. Are you aware of that? What do we have to adjust there?

**Comment by mhsdesign** [comment]
I havent checked but as im aware of the neos tests fail because of https://github.com/neos/flow-development-collection/pull/2900 this pr is to fix this regression...

lets see if the tests pass here: https://github.com/neos/neos-development-collection/pull/3943

**Comment by mhsdesign** [comment]
haha @mficzel youre right its not fixed. Private dindt do it protected doesnt either lets try public ^^
@kitsunet i guess you loose ^^

**Comment by mhsdesign** [comment]
lets try that: https://github.com/neos/flow-development-collection/pull/2965

**Comment by kitsunet** [comment]
unfortunate :D

**Pull request merged**